addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -70,17 +70,40 @@ weSnippetEditor.Class.include({
                 size: 'medium',
                 title: _t("Google Map API Key"),
                 buttons: [
-                    {text: _t("Save"), classes: 'btn-primary', close: true, click: async () => {
-                        const newAPIKey = dialog.$('#api_key_input').val() || false;
-                        await this._rpc({
-                            model: 'website',
-                            method: 'write',
-                            args: [
-                                [websiteId],
-                                {google_maps_api_key: newAPIKey},
-                            ],
-                        });
-                        invalidated = true;
+                    {text: _t("Save"), classes: 'btn-primary', click: async (ev) => {
+                        const $apiKeyInput = dialog.$('#api_key_input');
+                        const valueAPIKey = $apiKeyInput.val();
+                        const $apiKeyHelp = dialog.$('#api_key_help');
+                        if (!valueAPIKey) {
+                            $apiKeyInput.addClass('is-invalid');
+                            $apiKeyHelp.text(_t('Enter an API Key'));
+                            return;
+                        }
+                        const $button = $(ev.currentTarget);
+                        $button.prop('disabled', true);
+                        try {
+                            const response = await fetch(`https://maps.googleapis.com/maps/api/staticmap?center=belgium&size=10x10&key=${valueAPIKey}`);
+                            if (response.status === 200) {
+                                await this._rpc({
+                                    model: 'website',
+                                    method: 'write',
+                                    args: [
+                                        [websiteId],
+                                        {google_maps_api_key: valueAPIKey},
+                                    ],
+                                });
+                                invalidated = true;
+                                dialog.close();
+                            } else {
+                                const text = await response.text();
+                                $apiKeyInput.addClass('is-invalid');
+                                $apiKeyHelp.html(`${_t('Invalid API Key. The following error was returned by Google:')} <i>${text}</i>`);
```

Comment:
This will raise a CI/Security on runbot.
`text` is not escaped, it could inject code, see if return `<b>ok</b>`
![image](https://user-images.githubusercontent.com/30048408/122933944-5e308200-d36f-11eb-888a-d00fbd8f0d3d.png)
I know it's Google but still, it should [be escaped](https://www.odoo.com/documentation/14.0/developer/reference/security.html?highlight=sanitize#escaping-vs-sanitizing).
The easiest way here is probably to 
1. use `.empty().append()` to avoid the CI Security (since we take care of it, not just to avoid it..)
2. construct the `<i/>` in jquery and pass it the text in its text property to take care of escaping

```javascript
$apiKeyHelp.empty().text(
    _t('Invalid API Key. The following error was returned by Google:')
).append($('<i/>', {
    text: text
}));
```

![image](https://user-images.githubusercontent.com/30048408/122935512-ae5c1400-d370-11eb-911f-8bce44c45d85.png)